src/http/authentication/sig_auth.h

```diff
@@ -72,37 +72,41 @@ namespace ccf
       const std::shared_ptr<enclave::RpcContext>& ctx,
       std::string& error_reason) override
     {
-      const auto signed_request = parse_signed_request(ctx);
-      if (signed_request.has_value())
+      SignedReq signed_request;
+
+      try
+      {
+        signed_request = parse_signed_request(ctx);
+      }
+      catch (const std::exception& e)
+      {
+        error_reason = e.what();
+        return nullptr;
```

Comment:
Perhaps worth keeping the `std::optional` instead of `nullptr`s?

Reply:
As discussed, the function already returned a `nullptr` on error (the default view for the GitHub diff is confusing) so I'll leave this as it is.